src/zkevm_specs/evm_circuit/execution/push.py

```diff
@@ -16,9 +18,12 @@ def push(instruction: Instruction):
     is_pushed = instruction.continuous_selectors(num_pushed, 32)
     is_padding = instruction.continuous_selectors(num_padding, 32)
 
+    # Constrain stack write value must be zero for PUSH0.
+    instruction.constrain_zero(is_push0 * instruction.sum(value_le_bytes))
```

Comment:
Why is this necessary? When the opcode is PUSH0, then `num_pushed = 0`.
This means that the `is_pushed` selectors will be all zeroes (because `sum(is_pushed) = num_pushed`, from `instruction.continuous_selectors(num_pushed, 32)`.
If is_pushed is all zeroes, then each `value_le_bytes` is 0 from the loop below (which is checking all indexes of value_le_bytes: from 0 to 31).

Perhaps this was assuming that the loop was from 1 to 31?

Reply:
I see. I try to review this code again, the `value` has already constrained to zero of each byte for `PUSH0`.

I delete the redundant constraint in commit 01fd687. Thanks.